src/core/sprites/webgl/SpriteRenderer.js

```diff
@@ -317,20 +349,23 @@ export default class SpriteRenderer extends ObjectRenderer
             uint32View[index + 12] = uvs[2];
             uint32View[index + 17] = uvs[3];
 
-            uint32View[index + 3] = uint32View[index + 8] = uint32View[index + 13] = uint32View[index + 18] = tint;
-            float32View[index + 4] = float32View[index + 9] = float32View[index + 14] = float32View[index + 19] = textureId;
+            uint32View[index + 3] = uint32View[index + 8] = uint32View[index + 13] = uint32View[index + 18] = sprite._tintRGB + (sprite.worldAlpha * 255 << 24);
+
+            float32View[index + 4] = float32View[index + 9] = float32View[index + 14] = float32View[index + 19] = nextTexture._virtalBoundId;
 
             index += 20;
         }
 
         currentGroup.size = i - currentGroup.start;
 
-        this.vertexCount++;
+        // keep an eye out here!
+        // TODO - check this out for mobile!
 
+        /*
         if (this.vaoMax <= this.vertexCount)
```

Comment:
Why are we creating/storing the multiple VAOs if they are no longer used? Looks like only 1 is used now (since we only have 1 shader) so can we remove the loops creating these and change it to a single VAO like you did with shader?